chains/solana/contracts/programs/ccip-router/src/instructions/v1/token_admin_registry.rs

```diff
@@ -53,20 +58,87 @@ pub fn register_token_admin_registry_via_owner(
 }
 
 pub fn set_pool(
-    ctx: Context<ModifyTokenAdminRegistry>,
+    ctx: Context<SetPoolTokenAdminRegistry>,
     mint: Pubkey,
-    pool_lookup_table: Pubkey,
+    writable_indexes: Vec<u8>,
 ) -> Result<()> {
+    // set new lookup table
     let token_admin_registry = &mut ctx.accounts.token_admin_registry;
     let previous_pool = token_admin_registry.lookup_table;
-    token_admin_registry.lookup_table = pool_lookup_table;
+    let new_pool = ctx.accounts.pool_lookuptable.key();
+    token_admin_registry.lookup_table = new_pool;
+
+    // set writable indexes
+    // used to build offchain tokenpool accounts in a transaction
+    // indexes can be checked to indicate is_writable for the specific account
+    // this is also used to validate to ensure the correct write permissions are used according the admin
+    token_admin_registry.reset_writable();
+    for ind in writable_indexes {
+        token_admin_registry.set_writable(ind)
+    }
 
-    // TODO: Validate here that the lookup table has everything
+    // validate lookup table contains minimum required accounts if not zero address
+    if new_pool != Pubkey::zeroed() {
```

Comment:
nit: If we use `Pubkey::default()` it returns the same thing as `Pubkey::zeroed()` and can make you remove the dependency on `bytemuck::Zeroable` here
```suggestion
    if new_pool != Pubkey::default() {
```